provider/src/main/java/com.mindscapehq.android.raygun4android/services/CrashReportingPostService.java

```diff
@@ -0,0 +1,134 @@
+            SimpleDateFormat dateFormat = new SimpleDateFormat("yyyyMMddHHmmss");
+            String uuid = UUID.randomUUID().toString().replace("-", "");
+            String timestamp = dateFormat.format(new Date(System.currentTimeMillis()));
+            File fn = new File(getCacheDir(), timestamp + "-" + uuid + "." + RaygunSettings.DEFAULT_FILE_EXTENSION);
+
+            try {
+              SerializedMessage serializedMessage = new SerializedMessage(message);
+              ObjectOutputStream out = new ObjectOutputStream(new FileOutputStream(fn));
+              out.writeObject(serializedMessage);
+              out.close();
+            } catch (FileNotFoundException e) {
+              RaygunLogger.e("Error creating file when caching message to filesystem - " + e.getMessage());
+            } catch (IOException e) {
+              RaygunLogger.e("Error writing message to filesystem - " + e.getMessage());
+            }
+          } else {
+            RaygunLogger.w("Can't write crash report to local disk, maximum number of stored reports reached.");
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Raw post method that delivers a pre-built Crash Reporting payload to the Raygun API.
+   *
+   * @param apiKey      The API key of the app to deliver to
+   * @param jsonPayload The JSON representation of a RaygunMessage to be delivered over HTTPS.
+   * @return HTTP result code - 202 if successful, 403 if API key invalid, 400 if bad message (invalid properties), 429 if rate limited
+   */
+  private static int postCrashReporting(String apiKey, String jsonPayload) {
+    try {
+      if (validateApiKey(apiKey)) {
+        String endpoint = RaygunSettings.getCrashReportingEndpoint();
+        MediaType MEDIA_TYPE_JSON = MediaType.parse("application/json; charset=utf-8");
+
+        OkHttpClient client = new OkHttpClient.Builder()
+            .connectTimeout(NETWORK_TIMEOUT, TimeUnit.SECONDS)
+            .writeTimeout(NETWORK_TIMEOUT, TimeUnit.SECONDS)
+            .readTimeout(NETWORK_TIMEOUT, TimeUnit.SECONDS)
+            .build();
+
+        RequestBody body = RequestBody.create(MEDIA_TYPE_JSON, jsonPayload);
+
+        Request request = new Request.Builder()
+            .url(endpoint)
+            .header("X-ApiKey", apiKey)
+            .post(body)
+            .build();
+
+        Response response = null;
+
+        try {
+          response = client.newCall(request).execute();
+          RaygunLogger.d("Exception message HTTP POST result: " + response.code());
+          return response.code();
+        } catch (IOException ioe) {
+          RaygunLogger.e("OkHttp POST to Raygun Crash Reporting backend failed - " + ioe.getMessage());
+          ioe.printStackTrace();
+        } finally {
+          if (response != null) response.body().close();
+        }
+      }
+    } catch (Exception e) {
+      RaygunLogger.e("Can't postCrashReporting exception - " + e.getMessage());
```

Comment:
Should this be "Can't post to Crash Reporting. Exception - "? Similar text used in the RUMPostService and is a bit more clear.

Reply:
Yes it should be. Ill fix it up.